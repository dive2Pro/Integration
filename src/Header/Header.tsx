/**
 *
 */
import * as React from 'react'
import styled from 'styled-components'

const  Cookies =require( 'js-cookie')

const HeaderNav = styled.nav`
 display:flex;
 & > div{
 
   &:first-child{
     flex:1;
     text-align:start;
    }
  }
`

const AuthSection = styled.section`
&>div{
  margin-right:10px;
}
`

interface IHeaderState {
    user: { username: string } | undefined
}
interface  IHeaderProps{
    toggleLogin:()=>void
}

export const fetchSomething=()=> fetch("/api")

export default class Header extends React.Component<IHeaderProps, IHeaderState> {
    state = {user:undefined}

    componentWillMount() {

        let user = Cookies.get('auth')
        if(user){
        user = JSON.parse(user)
        this.setState({user})
        }

    }

    componentDidMount(){
       fetchSomething()
           .then(data=>{
               console.log(data.json())
           }).catch(err=>{
           console.error('err',err)
       })
    }

    handlerClickAuth=()=>{
        this.props.toggleLogin();
    }

    renderAuthDiv = (text: string) => {

        return (
            <div
                onClick={this.handlerClickAuth}
                id="sign">
                {text}
            </div>
        )
    }
    renderAuthed = () => {
        const user:any = this.state.user
        if(user==null){
            throw new Error('user cant be null');
        }
        const {username}=user
        return (
            <AuthSection>
                <div>My Polls</div>
                <div>New Poll</div>
                {this.renderAuthDiv(username)}
            </AuthSection>
        )
    }

    renderUnAuth = () => {
        return this.renderAuthDiv('Sign in/up')
    }

    render() {
        const isAuthed = !!this.state.user
        const Content = isAuthed ? this.renderAuthed : this.renderUnAuth
        return (
            <HeaderNav>
                <div>
                    fcc-voting
                </div>

                <div>
                    HOME
                </div>
                {
                    Content()
                }
            </HeaderNav>
        )
    }
}
