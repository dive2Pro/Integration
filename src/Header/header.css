nav {
    display: flex;

}

nav:first-child{
    flex:1;
}
