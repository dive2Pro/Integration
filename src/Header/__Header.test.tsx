/**
 *
 */
import * as React from 'react' 
import { shallow} from 'enzyme'
import Header
    from './Header'
import * as Cookie from 'js-cookie';

jest.mock('./Header',()=>{

})


 describe("Header",()=>{
     const cookieName = 'auth'
     beforeEach(()=>{
         const fc:any  =fetch
         fc.mockResponseOnce(
             JSON.stringify({
                 test:'test'
             })
         )
     })

     it('unauth render',()=>{
         // cookie
         Cookie.remove(cookieName)
         let  wrapper=shallow(<Header/>)
         expect(wrapper).toMatchSnapshot()
         expect(wrapper.find('div').length).toEqual(3)
         const signText = wrapper.find('#sign').text()
         expect(signText).toEqual("Sign in/up")
     })

     it('auth render',()=>{
         const user = {
             username:'hyc'
         }

         Cookie.set(cookieName,user)
         let  wrapper=shallow(<Header/>)
         expect(wrapper).toMatchSnapshot()
         expect(wrapper.find('div').length).toEqual(5)
         const signText = wrapper.find('#sign').text()
         expect(signText).toEqual(user.username)
         expect(wrapper.containsAllMatchingElements([
             <div>My Polls</div>,
             <div>New Poll</div>
         ])).toBeTruthy()
     })

     it('')

 })