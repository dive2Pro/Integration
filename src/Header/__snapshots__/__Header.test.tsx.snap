// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header auth click 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <styled.nav>
    <div>
        fcc-voting
    </div>
    <div>
        HOME
    </div>
    <styled.section>
        <div>
            My Polls
        </div>
        <div>
            New Poll
        </div>
        <div
            id="sign"
            onClick={[Function]}
        >
            hyc
        </div>
    </styled.section>
</styled.nav>,
  "nodes": Array [
    <styled.nav>
      <div>
            fcc-voting
      </div>
      <div>
            HOME
      </div>
      <styled.section>
            <div>
                  My Polls
            </div>
            <div>
                  New Poll
            </div>
            <div
                  id="sign"
                  onClick={[Function]}
            >
                  hyc
            </div>
      </styled.section>
</styled.nav>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Header
        toggleLogin={[Function]}
/>,
      "_debugID": 5,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Header {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handlerClickAuth": [Function],
        "props": Object {
          "toggleLogin": [Function],
        },
        "refs": Object {},
        "renderAuthDiv": [Function],
        "renderAuthed": [Function],
        "renderUnAuth": [Function],
        "state": Object {
          "user": Object {
            "username": "hyc",
          },
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 3,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <styled.nav>
          <div>
                    fcc-voting
          </div>
          <div>
                    HOME
          </div>
          <styled.section>
                    <div>
                              My Polls
                    </div>
                    <div>
                              New Poll
                    </div>
                    <div
                              id="sign"
                              onClick={[Function]}
                    >
                              hyc
                    </div>
          </styled.section>
</styled.nav>,
        "_debugID": 6,
        "_renderedOutput": <styled.nav>
          <div>
                    fcc-voting
          </div>
          <div>
                    HOME
          </div>
          <styled.section>
                    <div>
                              My Polls
                    </div>
                    <div>
                              New Poll
                    </div>
                    <div
                              id="sign"
                              onClick={[Function]}
                    >
                              hyc
                    </div>
          </styled.section>
</styled.nav>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Header
    toggleLogin={[Function]}
/>,
}
`;

exports[`Header auth render 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <styled.nav>
    <div>
        fcc-voting
    </div>
    <div>
        HOME
    </div>
    <styled.section>
        <div>
            My Polls
        </div>
        <div>
            New Poll
        </div>
        <div
            id="sign"
            onClick={[Function]}
        >
            hyc
        </div>
    </styled.section>
</styled.nav>,
  "nodes": Array [
    <styled.nav>
      <div>
            fcc-voting
      </div>
      <div>
            HOME
      </div>
      <styled.section>
            <div>
                  My Polls
            </div>
            <div>
                  New Poll
            </div>
            <div
                  id="sign"
                  onClick={[Function]}
            >
                  hyc
            </div>
      </styled.section>
</styled.nav>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Header
        toggleLogin={[Function]}
/>,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Header {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handlerClickAuth": [Function],
        "props": Object {
          "toggleLogin": [Function],
        },
        "refs": Object {},
        "renderAuthDiv": [Function],
        "renderAuthed": [Function],
        "renderUnAuth": [Function],
        "state": Object {
          "user": Object {
            "username": "hyc",
          },
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <styled.nav>
          <div>
                    fcc-voting
          </div>
          <div>
                    HOME
          </div>
          <styled.section>
                    <div>
                              My Polls
                    </div>
                    <div>
                              New Poll
                    </div>
                    <div
                              id="sign"
                              onClick={[Function]}
                    >
                              hyc
                    </div>
          </styled.section>
</styled.nav>,
        "_debugID": 4,
        "_renderedOutput": <styled.nav>
          <div>
                    fcc-voting
          </div>
          <div>
                    HOME
          </div>
          <styled.section>
                    <div>
                              My Polls
                    </div>
                    <div>
                              New Poll
                    </div>
                    <div
                              id="sign"
                              onClick={[Function]}
                    >
                              hyc
                    </div>
          </styled.section>
</styled.nav>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Header
    toggleLogin={[Function]}
/>,
}
`;

exports[`Header unauth render 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <styled.nav>
    <div>
        fcc-voting
    </div>
    <div>
        HOME
    </div>
    <div
        id="sign"
        onClick={[Function]}
    >
        Sign in/up
    </div>
</styled.nav>,
  "nodes": Array [
    <styled.nav>
      <div>
            fcc-voting
      </div>
      <div>
            HOME
      </div>
      <div
            id="sign"
            onClick={[Function]}
      >
            Sign in/up
      </div>
</styled.nav>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Header
        toggleLogin={[Function]}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Header {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handlerClickAuth": [Function],
        "props": Object {
          "toggleLogin": [Function],
        },
        "refs": Object {},
        "renderAuthDiv": [Function],
        "renderAuthed": [Function],
        "renderUnAuth": [Function],
        "state": Object {
          "user": undefined,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <styled.nav>
          <div>
                    fcc-voting
          </div>
          <div>
                    HOME
          </div>
          <div
                    id="sign"
                    onClick={[Function]}
          >
                    Sign in/up
          </div>
</styled.nav>,
        "_debugID": 2,
        "_renderedOutput": <styled.nav>
          <div>
                    fcc-voting
          </div>
          <div>
                    HOME
          </div>
          <div
                    id="sign"
                    onClick={[Function]}
          >
                    Sign in/up
          </div>
</styled.nav>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Header
    toggleLogin={[Function]}
/>,
}
`;
